csharp/src/Ice/CollocatedRequestHandler.cs

```diff
@@ -151,9 +151,14 @@ private async ValueTask InvokeAllAsync(OutgoingRequestFrame outgoingRequest, int
             {
                 if (_adapter.Communicator.TraceLevels.Protocol >= 1)
                 {
-                    // TODO we need a better API for tracing
-                    List<ArraySegment<byte>> requestData = Ice1Definitions.GetRequestData(outgoingRequest, requestId);
-                    TraceUtil.TraceSend(_adapter.Communicator, requestData);
+                    TraceFrame(_adapter.Communicator, requestId, outgoingRequest);
```

Comment:
I don't see any reason to create and call a local function here.

Reply:
This required to use stackallock , not allowed inside async method

Reply:
I think it would be cleaner to refactor the whole InvokeAllAsync method: not mark it async, do the sync processing first and finally call a local async method for the async processing.

Reply:
Maybe add a TODO?